src/hotspot/share/gc/shenandoah/heuristics/shenandoahOldHeuristics.cpp

```diff
@@ -124,6 +126,13 @@ bool ShenandoahOldHeuristics::prime_collection_set(ShenandoahCollectionSet* coll
     if (all_candidates_are_pinned()) {
       log_info(gc)("All candidate regions " UINT32_FORMAT " are pinned", unprocessed_old_collection_candidates());
       _old_generation->transition_to(ShenandoahOldGeneration::WAITING_FOR_FILL);
+    } else {
+      log_info(gc)("No regions selected for mixed collection. "
```

Comment:
When this fails, it may be useful to understand a bit more about why.  For example: heap->get_old_evac_reserve() is the evacuation budget.  Knowing how much is live (and how much garbage and "lost evacuation capacity" is in the rejected candidates (or in the first of the rejected candidates) might also be informative.